README.md

```diff
@@ -4,45 +4,149 @@
+of the resulting ensemble classifier on the test set.
+
+The values of the number of samples, number of features, and number of
+informative features can be controlled using command line arguments, as
+described in the help:
+
+```bash
+python demo.py blobs -h
+```
+
+### Digits data set
+
+The digits data is based on the well-known [handwritten digits data
+set](https://scikit-learn.org/stable/datasets/toy_dataset.html#optical-recognition-of-handwritten-digits-dataset).
+Each instance is a digitized image of a handwritten digit.  The images consist
+of 8x8 grayscale pixel values, which are represented as 64 features.  The goal
+is to construct a classifier that can predict the digit that is represented by
+an image.
+
+This demonstration constructs a binary classification problem using images for
+any two of the available ten digits, as specified by the `--digit1` and
+`--digit2` command line options (the defaults are 0 and 1).  As with the blobs
+data set, the available data are split into training and test sets.  The
+demonstration will print the number of features selected by QBoost, along with
+the prediction accuracy on the test set.
+
+The `--plot-digits` option is also provided to display one instance of the
+images for each of the two selected digits.  The displayed images are chosen
+randomly from the available data.
+
+The following command displays the available options for use with the digits
+data set:
+
+```bash
+python demo.py digits -h
+```
+
+## Code Overview
+
+Boosting methods construct a strong classifier by intelligently combining a set
+of weak classifiers.  Given a set of `N` weak classifiers, QBoost solves an
+optimization problem to select weak classifiers.  The objective of this
+optimization problem is to minimize the strong classifier's squared loss between
+actual and predicted targets.  A regularization term is also included to
+penalize complex models that include more weak classifiers.  The resulting
+optimization problem can be written as [1]:
+
+![Objective](images/objective.png)
+
+where `s` is an index over training instances, `i` is an index over features,
+`h_i(x_s)` denote the weak classifiers, `y_s` denote the observed targets, `w_i`
+are the weights to be determined, and lambda is the regularization parameter,
+which multiplies the L0 norm of the weight vector.  In this demonstration, the
+weights are treated as binary variables that take a value of either 0 or 1.  The
+weak classifiers are constructed from a series of single-feature decision tree
+rules, also known as decision stumps.  Following Refs. [1,4], the output of each
+weak classifier is scaled to `-1/N` and `+1/N`.
+
+By default, the demonstration is run with a fixed value of the regularization
+parameter, lambda.  If the `--cross-validation` option is used, then a simple
+parameter sweep is performed, and the value of lambda is selected on the basis
+of providing the highest accuracy on a validation set.
```

Comment:
Do you want to add here something about it taking several minutes?

Reply:
Added:

"This requires repeatedly
solving the optimization problem, and the demonstration may take several minutes
to run when this option is used."